app/cloud.go

```diff
@@ -0,0 +1,56 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+package app
+
+import (
+	"github.com/mattermost/mattermost-server/v5/model"
+)
+
+func (a *App) CheckAndSendUserLimitWarningEmails() *model.AppError {
+	if a.Srv().License() != nil && !*a.Srv().License().Features.Cloud {
+		// Not cloud instance, do nothing
+		return nil
+	}
+
+	// subscription, subErr := a.Cloud().GetSubscription()
+	// if subErr != nil {
+	// 	return subErr
+	// }
+
+	// if subscription != nil && subscription.IsPaidTier == "true" {
+	// 	// Paid subscription, do nothing
+	// 	return nil
+	// }
+
+	cloudUserLimit := *a.Config().ExperimentalSettings.CloudUserLimit
+	systemUserCount, _ := a.Srv().Store.User().Count(model.UserCountOptions{})
```

Comment:
I think this is not going to filter verified from non-verified. If we're sending the email only when the verification is done then we should filter by that

Reply:
I ran a test for this to be sure:
Put a log after this line to log the systemUserCount
User count was 10 when I started the test
Invited 5 people via email (and didn't verify their emails)
Invited a 6th person, and verified their email
The log for the verified email stated that the user count was 11 - so it seems to add up. 


There is an edge case here. User's created via an invite link (rather than an email) are created in the database _before_ their email is verified (whereas email invites are created _after_ their email is verified). The difference is, that an unverified email user created via link _does show in the Mattermost UI_ whereas the email invited one won't until its been both verified and created.
the third user in this image hasn't actually verified its email 
![image](https://user-images.githubusercontent.com/12176405/97018160-ceef7800-151c-11eb-807d-ab94637ab691.png)
I think we'll need to discuss when we consider a user to actually be a user, because it will affect a sysadmins decision to deactivate/activate certain users.

Reply:
@marianunez What are your thoughts on this?

Reply:
@ethervoid You can see discussion here - desired behaviour is in this branch! https://community.mattermost.com/private-core/pl/mm76dqizybrnujnboyj89tqpxh

Reply:
Cool! Good to have it cleared